test/e2e/suite/flake_attempt.go

```diff
@@ -0,0 +1,101 @@
+package suite
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	. "github.com/onsi/gomega"
+)
+
+type FlakeAttempt struct {
+	logger logr.Logger
+}
+type FlakeRun struct {
+	DeferCleanup    func(func(context.Context), ...interface{})
+	RetryableExpect func(actual interface{}, extra ...interface{}) Assertion
+}
+
+type IgnorablePanic struct{}
+
+func (IgnorablePanic) GinkgoRecoverShouldIgnoreThisPanic() {}
```

Comment:
do we need this anymore? it sees like when we want to panic with a retryable error, it doesn't matter what we panic with?